extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeChannelExtractor.java

```diff
@@ -168,107 +158,41 @@ public String getDescription() throws ParsingException {
 
-        JsonObject sectionListContinuation = ajaxJson.getObject(1).getObject("response")
-                .getObject("continuationContents").getObject("gridContinuation");
+        if (getTab("Videos") != null)
+            tabs.add(new YoutubeChannelVideosExtractor(getService(), (ListLinkHandler) getLinkHandler(), getTab("Videos")));
 
-        collectStreamsFrom(collector, sectionListContinuation.getArray("items"));
+        if (getTab("Playlists") != null)
+            tabs.add(new YoutubeChannelPlaylistsExtractor(getService(), (ListLinkHandler) getLinkHandler(), getName()));
 
-        return new InfoItemsPage<>(collector, getNextPageUrlFrom(sectionListContinuation.getArray("continuations")));
+        return tabs;
     }
 
-
-    private String getNextPageUrlFrom(JsonArray continuations) {
-        if (continuations == null) return "";
-
-        JsonObject nextContinuationData = continuations.getObject(0).getObject("nextContinuationData");
-        String continuation = nextContinuationData.getString("continuation");
-        String clickTrackingParams = nextContinuationData.getString("clickTrackingParams");
-        return "https://www.youtube.com/browse_ajax?ctoken=" + continuation + "&continuation=" + continuation
-                + "&itct=" + clickTrackingParams;
-    }
-
-    private void collectStreamsFrom(StreamInfoItemsCollector collector, JsonArray videos) throws ParsingException {
-        collector.reset();
-
-        final String uploaderName = getName();
-        final String uploaderUrl = getUrl();
-        final TimeAgoParser timeAgoParser = getTimeAgoParser();
-
-        for (Object video : videos) {
-            if (((JsonObject) video).getObject("gridVideoRenderer") != null) {
-                collector.commit(new YoutubeStreamInfoItemExtractor(
-                        ((JsonObject) video).getObject("gridVideoRenderer"), timeAgoParser) {
-                    @Override
-                    public String getUploaderName() {
-                        return uploaderName;
-                    }
-
-                    @Override
-                    public String getUploaderUrl() {
-                        return uploaderUrl;
-                    }
-                });
-            }
-        }
-    }
-
-    private JsonObject getVideoTab() throws ParsingException {
-        if (this.videoTab != null) return this.videoTab;
-
+    private JsonObject getTab(String tabName) throws ParsingException {
         JsonArray tabs = initialData.getObject("contents").getObject("twoColumnBrowseResultsRenderer")
                 .getArray("tabs");
         JsonObject videoTab = null;
 
         for (Object tab : tabs) {
             if (((JsonObject) tab).getObject("tabRenderer") != null) {
-                if (((JsonObject) tab).getObject("tabRenderer").getString("title").equals("Videos")) {
+                if (((JsonObject) tab).getObject("tabRenderer").getString("title").equals(tabName)) {
```

Comment:
Please use url in endpoint of tabrenderer instead of title, so that it won't break for other languages. See https://github.com/B0pol/NewPipeExtractor/commit/ffc22b798a2e82d5ffec1cc5b61ca214b5813a6a
You can use endsWith (maybe lowercase if needed) if you just wanna check, or split by "/" and pick last to gather it

Reply:
Thanks! I didn't think of localization.